src/WorkerChannel.ts

```diff
@@ -183,19 +193,34 @@ export class WorkerChannel implements IWorkerChannel {
         requestId: requestId,
         invocationResponse: response
       });
+
     }
 
     let { context, inputs } = CreateContextAndInputs(info, msg, logCallback, resultCallback);
     let userFunction = this._functionLoader.getFunc(<string>msg.functionId);
     
+    if (this.invocationRequestBefore) {
+      userFunction = this.invocationRequestBefore(context, userFunction);
+    }
+    
     // catch user errors from the same async context in the event loop and correlate with invocation
     // throws from asynchronous work (setTimeout, etc) are caught by 'unhandledException' and cannot be correlated with invocation
     try {
       let result = userFunction(context, ...inputs);
 
       if (result && isFunction(result.then)) {
-        result.then(result => (<any>context.done)(null, result, true))
-          .catch(err => (<any>context.done)(err, null, true));
+        result.then(result => {
+          if (this.invocationRequestAfter) {

```

Comment:
Are we able to call `this.invocationRequestAfter();` at the end of this whole function? Instead of putting multiple separate calls for possible outcomes of the result (regular, catching errors, catch outstanding errors).

Reply:
These are all happening in separate async paths which never converge on completion. I could see adding some `this.finish()` to each path to converge them. And within that we could call `invocationRequestAfter()`, but this would look pretty similar to what is already there. wdyt?

Reply:
`this.finish()` would be more elegant but not really a blocking issue :)